external/kafka/src/test/scala/org/apache/spark/streaming/kafka/KafkaRDDSuite.scala

```diff
@@ -92,8 +120,17 @@ class KafkaRDDSuite extends KafkaStreamSuiteBase with BeforeAndAfter {
       )
       until <- kc.getLatestLeaderOffsets(topicPartitions).right.toOption
     } yield {
-      KafkaRDD[String, String, StringDecoder, StringDecoder, String](
-        sc, kc.kafkaParams, from, until, mmd => s"${mmd.offset} ${mmd.message}")
+      val leaders = until.map { case (tp, lo) =>
```

Comment:
nit: Could you simplify this piece of code. It took me 5 mins to understand whats going on here. Its very Scala-esque and concise and elegant, I completely agree, but its still hard to read :)